components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -5788,7 +5788,8 @@ public void makeKeysForwardCompatible(ApiTypeWrapper apiTypeWrapper, String oldV
                             subscriptionStatus = APIConstants.SubscriptionStatus.ON_HOLD;
                         }
                         apiTypeWrapper.setTier(info.tierId);
-                        int subscriptionId = addSubscription(apiTypeWrapper, info.applicationId, subscriptionStatus);
+                        int subscriptionId = addSubscription(apiTypeWrapper, info.applicationId, subscriptionStatus,
+                                apiIdentifier.getProviderName());
```

Comment:
subscriber and provider are two different roles. `apiIdentifier.getProviderName()` is this the correct value to be set here?

Reply:
In this context API provider is the subscriber.